inst/include/sampler.h

```diff
@@ -156,7 +156,7 @@ namespace smc {
         ///Returns the history flags
         historyflags GetHistoryFlags(long n) const {return History[n].GetFlags();}
         /// Returns the Effective Sample Size of the specified particle generation.
-        long GetESS(long n) {return History[n].GetESS();}
+        const double GetESS(long n) {return History[n].GetESS();}
```

Comment:
At the risk of getting bogged down in trivia, I meant that this should be a const function [https://www.geeksforgeeks.org/const-member-functions-c/]
not that the return type needed a const qualifier, i.e. that it should be declared as
`double GettEss(long n) const`
because getters shouldn't change anything. Returning by const value is a bit esoteric and probably unnecessary.

C++ constness is fiddly...